share/namespace.go

```diff
@@ -203,6 +241,31 @@ func (n Namespace) Compare(n2 Namespace) int {
 	return bytes.Compare(n.data, n2.data)
 }
 
+// IsOutsideRange checks if the namespace is outside the min-max range of the given hashes.
+func (n Namespace) IsOutsideRange(leftHash, rightHash []byte) bool {
+	if len(leftHash) < NamespaceSize || len(rightHash) < 2*NamespaceSize {
```

Comment:
[question] how can the rightHash be 2 * the namespace size? I would expect leftHash and rightHash to be the same length and that length == NamespaceSize.

In this context, what is hash? I interpret it as the namespace of a left child or right child in an NMT.

Reply:
This is the root hash of the NMT, and IIRC, it has a length of 90 bytes.

Reply:
thanks! Not blocking but the param names are confusing. It seems like `leftHash` is expected to be the length of a namespace and rightHash is expected to be the length of two namespaces concatenated. Would it makes sense to rename:

- leftHash => leftNamespace
- rightHash => rootHash

perhaps with additional comments that describe the expected length of both

Reply:
I'm starting to think that `IsOutsideRange` will be useful only for the node. Maybe it's better to keep it there as a helper function?

Reply:
Works for me

Reply:
moved to node